compiler-core/src/parse/tests.rs

```diff
@@ -466,6 +466,52 @@ fn no_eq_after_binding1() {
     );
 }
 
+#[test]
```

Comment:
Can we also have a test for `echo 1 + 1` please 🙏

Reply:
Should that be the same as `echo { 1 + 1 }` or `{ echo 1 } + 1`?

Reply:
My intuition says `echo { 1 + 1 }` personally

Reply:
Mmh I would have said the opposite

Reply:
What about the operator precedence with other operators? Such as `echo 1 * 1` or `echo 1 == 1`?

Reply:
I would imagine echo to work a bit like let, where it has the lowest precedence. But I guess they are different things, since let is never used in sub-expressions. I would say for minimum confusion it should either have lowest precedence or highest precedence, so you can easily predict its behaviour

Reply:
Agreed. Either **highest** or **lowest**, but always consistent. My argument for **highest** is that it simplifies use cases. For example: Consider A) `echo 1 + 1` and B) `1 + echo 1`. If the echo keyword has the **lowest** precedence, the case A would output 2, equivalent to `{echo { 1 + 1 }}` while case B would output 1, equivalent to `1 + {echo { 1 }}`. However if it has the lowest precedence, then these cases would both output 1 and would both be equivalent to `{echo { 1 }} + 1` and `1 + {echo { 1 }}`.

Reply:
I guess if echo is considered to be a prefix operator, like `!` or `-`, it would make more sense to have higher precedence. But as a keyword, it feel like it should have lower precedence (e.g. `return` in other languages)

Reply:
I don't think it really makes sense comparing `echo` to `return`, seeing as `echo` is an expression and `return` in a statement.

Reply:
It should have the lowest so that you can easily add and remove `echo` to the start of the line and print the whole line. Having to wrap expressions in `{}` would be irritating.

> My argument for highest is that it simplifies use cases in sub-expressions. For example: Consider A) echo 1 + 1 and B) 1 + echo 1.

My intention was for `1 + echo 1` to be a syntax error, but perhaps I can be persuaded otherwise.

Reply:
I think it could be nice for echo to be allowed as an expression, like rust's `dbg!` macro

Reply:
Ok, I made `echo` lower precedence than binary operations!
Also @lpil we do have a test for `echo 1 + 1` here <https://github.com/gleam-lang/gleam/pull/3676/files#diff-35d6f2cca3b4773466d26264b7d213f97615c5de54ce0a5e7725593aca7e5207R485>

Reply:
Ok, so the expected behaviour is for `echo` to be an expression with the lowest possible precedence?

Reply:
Yeah, `echo 1 + 1` is the same as `echo { 1 + 1 }`

Reply:
Would then also `let a = echo 1 + 1` be equivalent too `let a = 1 + 1` then `echo a`?

Reply:
Yes it will